sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -121,7 +121,9 @@ class UDFRegistration private[sql] (functionRegistry: FunctionRegistry)
    */
   private[sql] def registerJavaUDAF(name: String, className: String): Unit = {
     try {
-      val clazz = Utils.classForName[AnyRef](className)
+      val clazz = session.artifactManager.withResources {
+        Utils.classForName[AnyRef](className, noSparkClassLoader = true)
```

Comment:
We could probably be a bit more direct here, by using the classloader produced by the artifactManager directly.